solution/ui/regulations/eregs-vite/src/components/search/RelatedSections.vue

```diff
@@ -1,67 +1,61 @@
 <script setup>
 import _groupBy from "lodash/groupBy";
 
-import { computed, ref } from "vue";
+import { inject } from "vue";
 
 import { locationLabel, locationUrl } from "utilities/filters";
 
 const props = defineProps({
-    base: {
-        type: String,
-        required: true,
-    },
     item: {
         type: Object,
         required: true,
     },
+    label: {
+        type: String,
+        default: "Related Section",
+    },
     partsLastUpdated: {
         type: Object,
-        required: true,
-    },
-    partsList: {
-        type: Array,
-        required: true,
+        default: () => {},
     },
 });
 
-const locationsCount = props.item.locations.length;
-const groupedLocations = _groupBy(props.item.locations, "title");
+const base = inject("base");
+
+const filteredLocations = props.item.locations.filter(
+    (location) => props.partsLastUpdated[location.part]
+);
```

Comment:
locations are now filtered here in a single location instead of in each page's view